src/main/kotlin/io/snyk/plugin/services/SnykTaskQueueService.kt

```diff
@@ -166,8 +179,41 @@ class SnykTaskQueueService(val project: Project) {
                         scanPublisher?.scanningIacError(iacResult.error!!)
                     }
                 }
+                LOG.info("IaC scan completed")
             }
-        }.queue()
+        })
+    }
+
+    private fun scheduleContainerScan() {
+        taskQueueIacContainer.run(object : Task.Backgroundable(project, "Snyk Container is scanning", true) {
+            override fun run(indicator: ProgressIndicator) {
+                val toolWindowPanel = project.service<SnykToolWindowPanel>()
+                if (toolWindowPanel.currentContainerResult != null) return
+
+                val iacResult = toolWindowPanel.currentIacResult
```

Comment:
It seems we use ToolWindowPanel also in other parts of this service. My doubts that ToolWindowPanel is ui part and we are tied hard background service with ui here. Maybe it will be better to save results in other place. But it's just my thoughts about this. Added here to not forget :)

Reply:
yes, you're right. At the moment this is *the way* how results are shared, but we have to refactor/modify the whole tree.